consensus/bor/bor.go

```diff
@@ -399,6 +399,22 @@ func (c *Bor) verifyCascadingFields(chain consensus.ChainHeaderReader, header *t
 		return err
 	}
 
+	if (number+1)%c.config.Sprint == 0 {
```

Comment:
It's already checking it from previous block's header and that header is built from the `BorValidatorSet` contract. https://github.com/maticnetwork/bor/blob/master/consensus/bor/bor.go#L403-L417

Reply:
the first block of a sprint will takes validators from last checkpoint, without validation.